core/src/main/scala/org/http4s/MediaType.scala

```diff
@@ -104,9 +88,34 @@ object MediaRange {
   /** Parse a MediaRange
     */
   def parse(s: String): ParseResult[MediaRange] =
-    new Http4sParser[MediaRange](s, "Invalid Media Range") with MediaRangeParser {
-      def main = MediaRangeFull
-    }.parse
+    ParseResult.fromParser(fullParser, "media range")(s)
+
+  private[http4s] val mediaTypeExtensionParser: Parser1[(String, String)] = {
+    import Parser.char
+    import Rfc2616BasicRules.optWs
+    import org.http4s.internal.parsing.Rfc7230.{quotedString, token}
+
+    val escapedString = "\\\\"
+    val unescapedString = "\\"
+
+    (char(';') *> optWs *> token ~ (char('=') *> token.orElse(quotedString)).?).map {
```

Comment:
optWs is from `Rfc2616BasicRules.optWs` we have a `ows` in `Rfc7230` as well, I suggest we use that one instead.

Reply:
This one is bloated enough, but I'll mop that up in a follow-up.